crates/sui-adapter/src/programmable_transactions/context.rs

```diff
@@ -391,43 +392,41 @@ impl<'vm, 'state, 'a, 'b, S: StorageView> ExecutionContext<'vm, 'state, 'a, 'b,
 
     /// Create a new package
     pub fn new_package<'p>(
-        &mut self,
-        modules: Vec<move_binary_format::CompiledModule>,
+        &self,
+        modules: &[CompiledModule],
```

Comment:
super nit (and maybe for the future): Why not also take these as an `impl IntoIterator<...>` for generality? Mainly because it feels a bit lopsided that these aren't passed in the same way as `dependencies`.

Reply:
I did try `impl Iterator` initially, but it doesn't work so well because we need to iterate over the modules twice -- once to build the type origin table, and another time to serialize them, so it wasn't a great fit (Might be possible with `impl Clone + Iterator<Item = ...>` but this starts to get a bit unwieldy, and there's only a couple of places (all in tests, I think) where we avoid spilling to a vector this way.